### ISSUE TITLE
Option to configure persistent caching for all cells within app or directory

### ISSUE THREAD

**Issue by leventov** [marimo-team/marimo#3054]
### Description

When the notebook is used locally always saving all results to disk is cheap. Therefore it makes sense to alleviate the cognitive burden for developers (assessing whether the cell is "expensive" or not) and source code clutter (that is, extra `@persistent_cache` decorators) by providing an option configurable per-app or per-directory (such as in a `.marimo/config` yaml/config file) to automatically save and invalidate the results of all cells whenever they are changed.

### Suggested solution

As described above.

### Alternative

An App` subclass, such as `PersistedApp`? But the problem with this is that it still has decision burden (per app rather than per cell, but still), and it cannot be dropped transparently when the app is deployed on the web.

**Comment by dmadisetti** [comment]
Yes! I have a working branch that does this. There are a few bugs, and will probably go into experimental `.toml` setting for now.

Can you expound more on what you mean by having persistent cache for a directory?

**Comment by gabrielgrant** [comment]
This would be great, so long as it is paired with:

1. Some indication of whether results are being computed or coming from cache
2. An easy way to clear/ignore the cache and recompute
